app/controllers/user_api_keys_controller.rb

```diff
@@ -51,22 +51,23 @@ def create
 
     require_params
 
-    unless SiteSetting.allowed_user_api_auth_redirects
+    if params.key?(:auth_redirect) && SiteSetting.allowed_user_api_auth_redirects
```

Comment:
yeah ... unless is just so hard to reason about ... agree with this change and sorry about using it there in the first place :)